controller/cache/dlp_rule.go

```diff
@@ -1149,35 +1143,21 @@ func (m *CacheMethod) GetAllDlpSensors(acc *access.AccessControl) []*api.RESTDlp
 						})
 					}
 					resp.RuleList = append(resp.RuleList, rdre)
-				} else { //try predefined rule
-					cdrelist := getPreDlpRuleFromDefaultSensor(cdrename)
-					for _, cdre := range cdrelist {
-						rdre := &api.RESTDlpRule{
-							Name:    common.GetOrigDlpRuleName(cdre.Name),
-							ID:      cdre.ID,
-							CfgType: cfgTypeMapping[cdre.CfgType],
-						}
-						for _, cpt := range cdre.Patterns {
-							if cpt.Context == "" {
-								cpt.Context = share.DlpPatternContextDefault
-							}
-							rdre.Patterns = append(rdre.Patterns, api.RESTDlpCriteriaEntry{
-								Key:     cpt.Key,
-								Value:   cpt.Value,
-								Op:      cpt.Op,
-								Context: cpt.Context,
-							})
-						}
-						resp.RuleList = append(resp.RuleList, rdre)
-					}
 				}
 			}
 		}
 		sort.Slice(resp.RuleList, func(i, j int) bool {
 			return resp.RuleList[i].Name < resp.RuleList[j].Name
 		})
-		ret = append(ret, &resp)
+		if getLocal && cdr.CfgType != share.FederalCfg {
```

Comment:
getLocal & getFed could be checked before line 1094,
```
resp := api.RESTDlpSensor{
			Name:      cdr.Name,
			GroupList: make([]string, 0),
			RuleList:  make([]*api.RESTDlpRule, 0),
			Comment:   cdr.Comment,
			Predefine: cdr.Predefine,
}
```
, to avoid unnecessary resp.GroupList and RuleList operation ?

Reply:
yes, I will fix it